app/models/instance_filter.rb

```diff
@@ -40,12 +46,33 @@ def scope_for(key, value)
     end
   end
 
+  def status_scope(value)
+    # The `join where` queries here look like they have a bug due to `domain_block`
+    # vs `domain_blocks`, however the table is `domain_blocks` while the relation on
+    # Instances is `domain_block`
+    case value.to_sym
+    when :allowed
+      Instance.joins(:domain_allow).reorder(Arel.sql('domain_allows.id desc'))
+    when :suspended
+      Instance.joins(:domain_block).where(domain_blocks: { severity: :suspend }).reorder(Arel.sql('domain_blocks.id desc'))
+    when :limited
+      Instance.joins(:domain_block).where(domain_blocks: { severity: [:silence, :noop] }).reorder(Arel.sql('domain_blocks.id desc'))
+    when :unrestricted
+      # Finds all instances where there isn't a record in the domain_blocks table
+      Instance.left_outer_joins(:domain_block).where(domain_blocks: { domain: nil })
```

Comment:
I think `missing` would work here...

```
mastodon(test)> Instance.left_outer_joins(:domain_block).where(domain_blocks: { domain: nil })
  Instance Load (0.3ms)  SELECT "instances".* FROM "instances" LEFT OUTER JOIN "domain_blocks" ON "domain_blocks"."domain" = "instances"."domain" WHERE "domain_blocks"."domain" IS NULL /* loading for pp */ LIMIT $1  [["LIMIT", 11]]
=> []
mastodon(test)> Instance.where.missing(:domain_block)
  Instance Load (0.7ms)  SELECT "instances".* FROM "instances" LEFT OUTER JOIN "domain_blocks" ON "domain_blocks"."domain" = "instances"."domain" WHERE "domain_blocks"."domain" IS NULL /* loading for pp */ LIMIT $1  [["LIMIT", 11]]
=> []
mastodon(test)> 
```

Reply:
That could work, I have no strong feelings here, just wanted to keep This reviewable